docs/components/VersionSelector.tsx

```diff
@@ -49,35 +50,42 @@ const STYLES_SELECT_ELEMENT = css`
 `;
 
 type Props = {
-  version: string;
-  onSetVersion: (value: string) => void;
   style?: React.CSSProperties;
 };
 
-const VersionSelector: React.FC<Props> = ({ version, style, onSetVersion }) => (
-  <div css={STYLES_SELECT} style={style}>
-    <label css={STYLES_SELECT_TEXT} htmlFor="version-menu">
-      <div>{Utilities.getUserFacingVersionString(version, LATEST_VERSION, BETA_VERSION)}</div>
-      <ChevronDownIcon style={{ height: '16px', width: '16px' }} />
-    </label>
-    {
-      // hidden links to help test-links spidering
-      VERSIONS.map(v => (
-        <a key={v} style={{ display: 'none' }} href={`/versions/${v}/`} />
-      ))
-    }
-    <select
-      id="version-menu"
-      css={STYLES_SELECT_ELEMENT}
-      value={version}
-      onChange={e => onSetVersion(e.target.value)}>
-      {VERSIONS.map(v => (
-        <option key={v} value={v}>
-          {Utilities.getUserFacingVersionString(v, LATEST_VERSION, BETA_VERSION)}
-        </option>
-      ))}
-    </select>
-  </div>
-);
+const VersionSelector: React.FC<Props> = ({ style }) => {
+  const { version, hasVersion, setVersion } = usePageApiVersion();
+
+  // Don't render the version selector on non-versioned pages
+  if (!hasVersion) {
+    return null;
+  }
+
+  return (
+    <div css={STYLES_SELECT} style={style}>
+      <label css={STYLES_SELECT_TEXT} htmlFor="version-menu">
+        <div>{Utilities.getUserFacingVersionString(version, LATEST_VERSION, BETA_VERSION)}</div>
+        <ChevronDownIcon style={{ height: '16px', width: '16px' }} />
+      </label>
+      {
+        // hidden links to help test-links spidering
```

Comment:
what does this comment mean?

in general, I think we should use comments to explain one-off exceptional behavior (like a comment explaining that Safari renders something incorrectly, so adding a style to fix it). With that, I don't think we need comments in most places. If it's not clear what this code does, I wonder if we could add a variable name to help explain.

Reply:
This is from the current version selector code. I didn't write it, but I believe this influenced SEO. By adding hidden links to all versions, the spider should easily pick up multiple versions. Without these hidden links, we don't have any direct URL reference to the other Expo SDK versions.

Definitely can improve that comment though.

Reply:
```suggestion
        // Add hidden links to create crawler references to all other SDK versions
        // This should help search bots help find SDK versions, while using JS to switch between them